pkgs/by-name/ap/apkeditor/package.nix

```diff
@@ -0,0 +1,172 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+
+  jre,
+  gradle_6,
+  makeWrapper,
+}:
+
+let
+  gradle = gradle_6;
+  REAndroidLibrary =
+    args:
+    let
+      inherit (args) pname version projectName;
+      outJar = "share/${projectName}/${projectName}.jar";
+      self = stdenv.mkDerivation (
+        {
+          __darwinAllowLocalNetworking = true;
+
+          buildInputs = [ jre ];
+          nativeBuildInputs = [ gradle ];
+
+          gradleFlags = [ "-Dfile.encoding=utf-8" ];
+          gradleBuildTask = "jar";
+          doCheck = true;
+
+          inherit outJar;
+          installPhase = ''
+            runHook preInstall
+            install -Dm644 build/libs/*.jar $out/${outJar}
+            runHook postInstall
+          '';
+        }
+        // args
+        // {
+          meta = {
+            sourceProvenance = with lib.sourceTypes; [
+              fromSource
+              binaryBytecode # mitm cache
+            ];
+          } // args.meta;
+        }
+      );
+    in
+    self;
+
+  arsclib = REAndroidLibrary {
+    pname = "arsclib";
+    version = "1.3.5-unstable-2024-10-21"; # 1.3.5 is not new enough for APKEditor
+    projectName = "ARSCLib";
+
+    src = fetchFromGitHub {
+      owner = "REAndroid";
+      repo = "ARSCLib";
+      rev = "ed6ccf00e56d7cce13e8648ad46a2678a6093248";
```

Comment:
Please provide a method for how you deduce these. Either document them and/or write an updater to do the update automatically.

Reply:
What I meant was documentation on how you decide which commit of each dep you used and which ones should be used in the future when updating.

Reply:
That's much more like it, thanks :)